plugins/destination/file/client/spec.go

```diff
@@ -0,0 +1,34 @@
+package client
+
+import (
+	"fmt"
+)
+
+type FormatType string
+
+type BackendType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type Spec struct {
+	Directory string     `json:"directory,omitempty"`
+	Format    FormatType `json:"format,omitempty"`
```

Comment:
Perhaps not for this first iteration, but I think we should also support:
 - a "max size" option so that files are guaranteed to be below that size, and a new file with a different suffix is created when the limit is reached
 - a setting to allow each table to be written to a directory of its own name. e.g. `aws_ec2_instances` should get written to `output/aws_ec2_instances/aws_ec2_instances.csv`, potentially with a way to specify a suffix so that different syncs don't overwrite each other (maybe sync time could be used). This will make it easier to upload to some destinations like S3/Athena (at least until we support those directly, but even then there will be other similar destinations we don't support)

Again, just some ideas while I'm thinking about this - it might require some more thought and design so doesn't need to be in this PR

Reply:
Yeah it can be good improvements but I think `max_size` should be on the SDK level at least for managed as they can choose either `size` or `batch_size` or `min(max_batch_size, max_size)`